meeteval/der/md_eval.py

```diff
@@ -0,0 +1,175 @@
+import logging
+import re
+import decimal
+import shutil
+import tempfile
+import subprocess
+import dataclasses
+from pathlib import Path
+
+import meeteval.io
+from meeteval.wer.wer.error_rate import ErrorRate
+
+
+def _fix_channel(r):
+    return meeteval.io.rttm.RTTM([
+        line.replace(channel='1')  # Thilo puts there usually some random value, e.g. <NA> for hyp and 0 for ref, while common default is 1
+        for line in r
+    ])
+
+
+@dataclasses.dataclass(frozen=True)
+class DiaErrorRate:
+    """
+
+    """
+    error_rate: 'float | decimal.Decimal'
+
+    scored_speaker_time: 'float | decimal.Decimal'
+    missed_speaker_time: 'float | decimal.Decimal'  # deletions
+    falarm_speaker_time: 'float | decimal.Decimal'  # insertions
+    speaker_error_time: 'float | decimal.Decimal'  # substitutions
+
+    @classmethod
+    def zero(cls):
+        return cls(0, 0, 0, 0, 0)
+
+    def __post_init__(self):
+        assert self.scored_speaker_time >= 0
+        assert self.missed_speaker_time >= 0
+        assert self.falarm_speaker_time >= 0
+        assert self.speaker_error_time >= 0
+        errors = self.speaker_error_time + self.falarm_speaker_time + self.missed_speaker_time
+        error_rate = decimal.Decimal(errors / self.scored_speaker_time)
+        if self.error_rate is None:
+            object.__setattr__(self, 'error_rate', error_rate)
+        else:
+            # decimal.Decimal.quantize rounds to the same number of digits as self.error_rate has.
+            error_rate = error_rate.quantize(self.error_rate)
+            assert self.error_rate == error_rate, (error_rate, self)
+
+    def __radd__(self, other: 'int') -> 'ErrorRate':
+        if isinstance(other, int) and other == 0:
+            # Special case to support sum.
+            return self
+        return NotImplemented
+
+    def __add__(self, other: 'DiaErrorRate'):
+        if not isinstance(other, self.__class__):
+            raise ValueError()
+
+        return DiaErrorRate(
+            error_rate=None,
+            scored_speaker_time=self.scored_speaker_time + other.scored_speaker_time,
+            missed_speaker_time=self.missed_speaker_time + other.missed_speaker_time,
+            falarm_speaker_time=self.falarm_speaker_time + other.falarm_speaker_time,
+            speaker_error_time=self.speaker_error_time + other.speaker_error_time,
+        )
+
+
+def md_eval_22(
```

Comment:
Can you split this into a Python function that takes `RTTM` objects and the cli function that takes file paths?

For WERs, the cli functions are defined in `__main__.py`.